### ISSUE TITLE
BoxView on Android does not constrain its width to container

### ISSUE THREAD

**Issue by activa** [dotnet/maui#17992]
### Description

On Android, when using a BoxView control inside a container, the boxview control will not use the container width and ignores HorizontalOptions:

```
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="BoxViewBug.MainPage">

    <Grid HorizontalOptions="Start" BackgroundColor="Red">

        <VerticalStackLayout>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>

        </VerticalStackLayout>
    </Grid>
</ContentPage>
```

iOS: 
![image](https://github.com/dotnet/maui/assets/650710/95723c0e-68a4-4d6c-bcd1-be69de752d35)
Windows:
![image](https://github.com/dotnet/maui/assets/650710/703c4bfb-f6b7-4435-9895-ff613b208978)
Android:
![image](https://github.com/dotnet/maui/assets/650710/780fe5a1-c78d-445f-a531-a2ab6da3cc86)

### Steps to Reproduce

Create the following page in a MAUI app:

```
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="BoxViewBug.MainPage">

    <Grid HorizontalOptions="Start" BackgroundColor="Red">

        <VerticalStackLayout>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>
            <Label Text="Hello, World!" />
            <BoxView HeightRequest="5" Color="Green"/>

        </VerticalStackLayout>
    </Grid>
</ContentPage>
```

Note that on Android, the Grid container is expanded to the full width of the screen. This is caused by the BoxView controls. It works as expected on iOS and Windows.
Worth noting is that this worked fine in MAUI 6.x and also Xamarin Forms

### Link to public reproduction project repository

https://github.com/activa/maui-boxview-bug

### Version with bug

7.0.96

### Is this a regression from previous behavior?

This used to work in MAUI .NET 6 and also in Xamarin.Forms

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

All Android version

**Comment by activa** [comment]
Looks like this was fixed by #6206
But it was not backported to 7.0

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.8.0 Preview 3.0(8.0.0-rc.2.9373). This issue does not repro on Android 13.0-API33, Windows 11 and iOS 16.4 with below Project:
[17992.zip](https://github.com/dotnet/maui/files/12924645/17992.zip)

Android:
![image](https://github.com/dotnet/maui/assets/97647558/281a58a7-6ae1-4407-bb47-b8aee3faef40)

**Comment by activa** [comment]
The latest "public" version is .NET 7. It works on .NET 8, but that's still in preview.

As I said in my original comment, this is broken in 7.0.96 because the fix was not backported to 7.0

**Comment by activa** [comment]
This is working fine in MAUI 8. It's just unfortunate that this was not fixed in .NET MAUI 7
